### ISSUE TITLE
Feature Request: Seeded test / train split

### ISSUE THREAD

**Issue by samtriolo** [benfred/implicit#409]
### Summary
Add a "seed" parameter to [implicit.evaluation.train_test_split()](https://github.com/benfred/implicit/blob/master/implicit/evaluation.pyx). This would have a couple uses, but one of the most appealing in my mind is when you want to better measure / compare / combine two datasets that leverage the same "users" and "items," but different "metrics." If there is another / existing way to achieve this or I'm otherwise mistaken, please let me know.

### Example
For example, if you are applying implicit to video pages on a website, however you want to better measure / compare / combine two metrics: how many times a user visited a video (ie each page load counts as +1) vs how many seconds of a video a user viewed.  You can more easily compare and potentially combine these two metrics (to eg produce a single score based on both) if they use the same train and test datasets, which can be ensured with a split seed.

### Implementation suggestion
If I'm not mistaken, you can simply add the parameter `seed=None` to the existing `train_test_split()` method, and then add code similar to the following near the top of the method:
```
    if seed:
        log.info(f'Using split seed {seed}')
        np.random.seed(seed)
```

**Comment by ita9naiwa** [comment]
#411
